core/vm/contracts.go

```diff
@@ -176,32 +173,39 @@ func init() {
 	for k := range PrecompiledContractsFjord {
 		PrecompiledAddressesFjord = append(PrecompiledAddressesFjord, k)
 	}
-	for k := range PrecompiledContractsCel2 {
-		PrecompiledAddressesCel2 = append(PrecompiledAddressesCel2, k)
-	}
-	for k := range PrecompiledCeloContractsCel2 {
-		PrecompiledAddressesCel2 = append(PrecompiledAddressesCel2, k)
-	}
 }
 
 // ActivePrecompiles returns the precompiles enabled with the current configuration.
 func ActivePrecompiles(rules params.Rules) []common.Address {
```

Comment:
Instead of changing this function, we could also rename it and wrap it in a new `ActivePrecompiles`:

```
func ActivePrecompiles(rules params.Rules) []common.Address {
    addresses := OptimismActivePrecompiles(rules)
	if !rules.IsCel2 {
		return addresses
	}

	PrecompiledAddressesCel2 = PrecompiledAddressesCel2[:0]
	PrecompiledAddressesCel2 = append(PrecompiledAddressesCel2, addresses...)

	for k := range PrecompiledCeloContractsCel2 {
		PrecompiledAddressesCel2 = append(PrecompiledAddressesCel2, k)
	}

	return PrecompiledAddressesCel2
}
```

The would keep our changes more clearly separated from the upstream code, so that we will get less conflicts when the next forks are added.

Reply:
Nice suggestion, updated.